doc/refactoring_design.md

```diff
@@ -0,0 +1,481 @@
+uint256 public constant TOTAL_SUPPLY = 1400000000 * 10 ** 18;    // OceanToken total supply
+
+// transfer tokens 
+function transfer(address _to, uint256 _value) public returns (bool);
+
+// mint new Ocean tokens and send tokens to block reward contract 
+function mintTokens() public returns (bool success);
+```
+
+
+### 2.2 Block Reward
+
+Block rewards are newly minted Ocean tokens which reward providers of data commons. 
+
+<img src="img/refactoring/blockreward.jpg" width="1000" />
+
+**Key Points**:
+
+* Block reward Ocean tokens will be minted and deposited into a reward pool;
+* In the same block interval, Ocean network randomly select a provider to serve the service agreement of data commons. Therefore, all providers have the equal probability to receive the block rewards.
+* When the service agreement is fulfilled and verified, provider receive a `lottery ticket` to win the block rewards;
+* In the end of current block interval, Ocean network randomly choose a winner from the providers who hold `lottery ticket`.
+* This chosen provider receives all tokens in the reward pool;
+* The procedure will repeat for each block interval; if there is no winner, Ocean tokens in the reward pool will accumulate over time until a winner claims the reward.
+
+<img src="img/refactoring/rewardLogic.jpg" width="1000" />
+
+The process within one block interval can be illustrated as below:
+
+<img src="img/refactoring/distributeBR.jpg" width="800" />
+
+
+
+The sample smart contract looks like this:
+
+```Solidity
+// providers of data commons in this block interval
+address[] candidates;
+
+// number of reward tokens in this block interval
+uint256 rewardPool;
+
+// randomly pick a provider to receive block reward
+function sendBlockReward() public returns (bool success){
+	...
+	// Ocean token contract mints new tokens
+	OceanTokenContract.mintTokens();
+	
+	// randomly pick a provider from the candidates (need a secure Random Number Generator!!!)
+	uint256 random = keccak256(abi.encodePacked(...)) % n;
+	
+	// send tokens in rewardPool to a randomly chosen provider
+	transfer(candidates[random], rewardPool);
+	...
+};
+```
+
+
+### 2.3 Service Agreement
+Please see [SLA.md](https://github.com/oceanprotocol/dev-ocean/blob/feature/SLA-specs/doc/sla.md) and [smart contract prototype](https://github.com/oceanprotocol/keeper-contracts/blob/develop/contracts/SLA/ServiceAgreement.sol) for details. 
+
+The structre of a Serivce Agreement looks like below:
+
+<img src="img/refactoring/SAStructure.jpg" width="700" />
+
+The key point is to split up the conditions from reward payment as:
+<img src="img/refactoring/SAOverview.jpg" width="1000" />
+
+
+**Some changes to be added:**
+
+
+1. **`Time Lock` and `Time Out`:** Each service condition has `time-lock` and `time-out`. It can only be fulfilled within the time window defined by the starting time point (`time-lock`) and the ending time point (`time-out`).
```

Comment:
This part is already implemented check out `timeoutValues` in the current [implementation](https://github.com/oceanprotocol/keeper-contracts/blob/develop/contracts/SLA/ServiceAgreement.sol#L31)

Reply:
Thanks for letting me know.